oci/oci.go

```diff
@@ -500,18 +513,20 @@ func (r *Runtime) UpdateStatus(c *Container) error {
 			if err != nil {
 				return fmt.Errorf("status code conversion failed: %v", err)
 			}
-			c.state.ExitCode = int32(statusCode)
+			c.State.ExitCode = int32(statusCode)
 		}
 	}
 
+	c.toDisk()
```

Comment:
@runcom 
> what I'm really doing here is storing c.State which contains fields like Started/Finished/Stopped

If I understand `toDisk()` correctly, you're unconditionally storing the whole container structure.
If we're only concerned with storing the state changes, we could easily do so from `UpdateStatus()` **only** when the container state actually changes.


> are they wasting so much IO as well ?

They seem to believe disk I/O is free, which I believe is a wrong assumption :) The impact may be marginal, I don't know, and this may be a micro-optimization. Problems arise when everyone makes the same assumption and macro benchmarks start showing non marginal disk I/O usage.